### DISCUSSION TITLE
ETABS & Python

### DISCUSSION THREAD

**Discussion by bstbstie** [xcfem/xc#127] [Q&A] [answered]
Hi there! I tried to get the maximum joint displacement but only the values under Ux, Uy, and Uz are all wrong. How can I reach those displacement values? Here is my code:

import comtypes
import comtypes.client
import win32com.client

def launch_and_open_etabs(file_path):
    try:
        my_etabs_object = win32com.client.Dispatch("CSI.ETABS.API.ETABSObject")
        my_etabs_object.ApplicationStart()
        my_etabs_model = my_etabs_object.SapModel
        ret = my_etabs_model.InitializeNewModel()  # You can also use InitializeNewModel() for a new model
        ret = my_etabs_model.File.OpenFile(file_path)

        if ret == 0:
            print(f"Successfully opened the ETABS file: {file_path}")
        else:
            print(f"Failed to open the ETABS file: {file_path}")
    
    except Exception as e:
        print(f"Error occurred: {str(e)}")

etabs_file_path = r"C:\ALL_FILES\03_PhD\PhD_Studies\06_4.TIK\Python_ETABS-API\ETABS\SalesforceT_1-story.edb"
launch_and_open_etabs(etabs_file_path)

ETABSObject=comtypes.client.GetActiveObject("CSI.ETABS.API.ETABSObject")
SapModel=ETABSObject.SapModel
ret_islocked=SapModel.SetModelIsLocked(False) # açık olan projenin kilidini açtık, true yazsaydık kilitleyecekti
ret_analysis=SapModel.Analyze.RunAnalysis() # run analysis

NumberTables=1
TableKey=[]
ImportType=[]
TableName=[]

availableTables=SapModel.DatabaseTables.GetAvailableTables(NumberTables, TableKey, TableName, ImportType) # table parameters

for i in range(availableTables[0]):
    print(f"{1}:{availableTables[1][i]}") # tabloları alt alta liste gibi yazdır

TableKey="Joint Displacements" 
FieldKeyList=[]
GroupName="All"
TableVersion=1
FieldKeyIncluded=[]
NumberRecords=1
Tabledata=[]

JointDisplacements=SapModel.DatabaseTables.GetTableforDisplayArray (TableKey, FieldKeyList, GroupName, TableVersion, FieldKeyIncluded, NumberRecords, Tabledata)

columns=JointDisplacements[4] 
print(columns[:100])

**Comment by lcpt** [comment, marked as answer]
Hi, @bstbstie.

Your question is related with ETABS, not with XC, right? I think you'll need to contact your ETABS support. If I'm wrong, please let me know.

**Reply by bstbstie** [reply]
Hi, @lcpt
Yes, I forgot to say that it is related to Etabs. Actually I found out that the problem was related to the units. By setting the correct units my problem is solved. Thank you so much for your interest 🙏🙏🙏
